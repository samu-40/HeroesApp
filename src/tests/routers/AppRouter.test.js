import '../../setupTest';
import { mount } from "enzyme";
import AppRouter from "../../routers/AppRouter";
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en AppRouter.js', () => {

    test('Mostrar <LoginScreen/> si no estoy autenticado ', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: { logged: false }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar <MarvelScreen/> si estoy autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: { 
                logged: true,
                name: 'Samuel'
            }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper.find('.navbar').exists()).toBe(true);

    });    

});
