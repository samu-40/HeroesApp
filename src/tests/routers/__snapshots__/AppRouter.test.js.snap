// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter.js Mostrar <LoginScreen/> si no estoy autenticado  1`] = `
<div>
  <div
    className="box"
  >
    <h1>
      Login
    </h1>
    <button
      className="btn"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
